runtime-transaction/benches/process_compute_budget_instructions.rs

```diff
@@ -0,0 +1,168 @@
+                        )
+                    })
+                    .collect();
+                let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+                bencher.iter(|| {
+                    (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                        assert!(process_compute_budget_instructions(black_box(
+                            tx.message().program_instructions_iter()
+                        ))
+                        .is_ok())
+                    })
+                });
+            },
+        );
+}
+
+fn bench_process_compute_budget_instructions_compute_budgets(c: &mut Criterion) {
+    c.benchmark_group("bench_process_compute_budget_instructions_compute_budgets")
+        .throughput(Throughput::Elements(NUM_TRANSACTIONS_PER_ITER as u64))
+        .bench_function("4 compute-budget instructions", |bencher| {
+            let ixs = vec![
+                ComputeBudgetInstruction::request_heap_frame(40 * 1024),
+                ComputeBudgetInstruction::set_compute_unit_limit(u32::MAX),
+                ComputeBudgetInstruction::set_compute_unit_price(u64::MAX),
+                ComputeBudgetInstruction::set_loaded_accounts_data_size_limit(u32::MAX),
+            ];
+            let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+            bencher.iter(|| {
+                (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                    assert!(process_compute_budget_instructions(black_box(
+                        tx.message().program_instructions_iter()
+                    ))
+                    .is_ok())
+                })
+            });
+        });
+}
+
+fn bench_process_compute_budget_instructions_builtins(c: &mut Criterion) {
+    c.benchmark_group("bench_process_compute_budget_instructions_builtins")
+        .throughput(Throughput::Elements(NUM_TRANSACTIONS_PER_ITER as u64))
+        .bench_function("4 dummy builtins", |bencher| {
+            let ixs = vec![
+                Instruction::new_with_bincode(solana_sdk::bpf_loader::id(), &(), vec![]),
+                Instruction::new_with_bincode(solana_sdk::secp256k1_program::id(), &(), vec![]),
+                Instruction::new_with_bincode(
+                    solana_sdk::address_lookup_table::program::id(),
+                    &(),
+                    vec![],
+                ),
+                Instruction::new_with_bincode(solana_sdk::loader_v4::id(), &(), vec![]),
+            ];
+            let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+            bencher.iter(|| {
+                (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                    assert!(process_compute_budget_instructions(black_box(
+                        tx.message().program_instructions_iter()
+                    ))
+                    .is_ok())
+                })
+            });
+        });
+}
+
+fn bench_process_compute_budget_instructions_mixed(c: &mut Criterion) {
+    let num_instructions = 1024;
```

Comment:
this is unrealistic though - we cannot have this many instructions given packet size. For *empty* ixs like the dummy ixs you have, the maximum ixs for a valid packet works out to 355 IIRC.

Fine if you don't map out the exact number, but I think we should at least be reasonably close to the actual worst-case instead of 3x larger, wdyt?

Reply:
I did want to stretch the limit, but yea, it's reasonable to stay close to reality.

Reply:
c5ca9784903177641efb06586a9669d5e147924d